symbolic_pymc/tensorflow/printing.py

```diff
@@ -42,19 +42,19 @@ def println(self, obj):
         self.buffer.write(self.format(obj) + "\n")
         self.buffer.flush()
 
-    def depth_check(self, current_depth):
+    def depth_check(self):
         if isinstance(self.depth_index, (list, tuple, range)):
-            return current_depth in self.depth_index
+            return self.depth_tracking.count("|") in self.depth_index
```

Comment:
Track the depth separately as a count; this approach is unnecessarily format-dependent and computationally wasteful.  

Plus, if you do it that way, it would be much easier to change the output so that a subgraph can be displayed as if it were the root of the graph.

Reply:
So the only way to specify a graph depth lower and upper bound is to use `range(0, upper_bound)` and `range(lower_bound, sys.maxsize)`?  That doesn't sound as simple as having, say, only upper and lower bound depth parameters.

Reply:
When specifying graph depth one doesn't have to use `range`, there is also the option to use a `list` or `tuple` object. I updated the function with a docstring with an example.